server/controller/src/main/java/ai/starwhale/mlops/domain/job/step/status/StepStatusMachine.java

```diff
@@ -24,32 +24,33 @@
 import static ai.starwhale.mlops.domain.job.step.status.StepStatus.READY;
 import static ai.starwhale.mlops.domain.job.step.status.StepStatus.RUNNING;
 import static ai.starwhale.mlops.domain.job.step.status.StepStatus.SUCCESS;
-import static ai.starwhale.mlops.domain.job.step.status.StepStatus.TO_CANCEL;
 import static ai.starwhale.mlops.domain.job.step.status.StepStatus.UNKNOWN;
 
 import java.util.Map;
 import java.util.Set;
-import org.springframework.stereotype.Component;
 
-@Component
 public class StepStatusMachine {
 
     static final Map<StepStatus, Set<StepStatus>> transferMap = Map.of(
-            CREATED, Set.of(READY, PAUSED, RUNNING, SUCCESS, TO_CANCEL, CANCELED, FAIL),
-            READY, Set.of(PAUSED, RUNNING, SUCCESS, TO_CANCEL, FAIL),
+            CREATED, Set.of(READY, PAUSED, RUNNING, CANCELLING, CANCELED, SUCCESS, FAIL),
+            READY, Set.of(PAUSED, RUNNING, CANCELLING, CANCELED, SUCCESS, FAIL),
             PAUSED, Set.of(READY, RUNNING, CANCELLING, CANCELED, FAIL),
             RUNNING, Set.of(CANCELLING, CANCELED, PAUSED, SUCCESS, FAIL),
-            SUCCESS, Set.of(), FAIL, Set.of(),
-            TO_CANCEL, Set.of(CANCELLING, CANCELED, FAIL),
-            CANCELLING, Set.of(CANCELED, FAIL), CANCELED, Set.of(),
-            UNKNOWN, Set.of(StepStatus.values()));
+            CANCELLING, Set.of(CANCELED, FAIL),
+            SUCCESS, Set.of(),
+            FAIL, Set.of(), //TODO
+            CANCELED, Set.of(),
+            UNKNOWN, Set.of(StepStatus.values())
+    );
 
-    public boolean couldTransfer(StepStatus statusNow, StepStatus statusNew) {
+    public static final Set<StepStatus> FINAL_STATUS = Set.of(FAIL, SUCCESS, CANCELED);
+
+    public static boolean couldTransfer(StepStatus statusNow, StepStatus statusNew) {
         return transferMap.get(statusNow).contains(statusNew);
     }
 
-    public boolean isFinal(StepStatus status) {
-        return transferMap.get(status).isEmpty();
+    public static boolean isFinal(StepStatus status) {
```

Comment:
How to mock a StepStatusMachine when we do UT if you change it to static

Reply:
see: https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#48